docs/notes.md

```diff
@@ -878,5 +881,438 @@ With these observations, we can simplify the combined \\(m\\)-party statement ab
 \\]
 
+Arithmetic Circuit Proofs
+=========================
+
+The goal of an *arithmetic circuit proof* is for a prover to convince a verifier that a particular set of values \\(v\\) satisfy the constraints represented by the arithmetic circuit, without revealing any additional information about the values \\(v\\).
+
+The prover will use the efficient inner product protocol to do this, so we want to work towards expressing the arithmetic circuit's conditions in terms of a single inner product. The prover begins with a vector of secret values \\(v\\), and a vector of commitments to those secret values \\(V\\). The prover will send \\(V\\) to the verifier, along with the proof.
+
+Notation for arithmetic circuit proofs
+------------------------------------------
+
+In the paper, matrices are labeled as \\( \textbf{W}\_L, \textbf{W}\_R, \textbf{W}\_O, \textbf{W}\_V \\). We will keep this notation, but will note to readers not to confuse the \\(\textbf{W}\_{L,R,O,V}\\) notation for being a vector of points. 
+
+We will use the notation described in the [`notation`](index.html#notation) section of the notes. In addition, we rename one more variable from the paper, to make it clear which variables are blinding factors:
+\\[
+\begin{aligned}
+    \beta         &\xrightarrow{} \tilde{o} \\\\
+\end{aligned}
+\\]
+
+Proving statements about arithmetic circuits gates
+--------------------------------------------------
+
+One type of gate in an arithmetic circuit is a multiplication gate of fan-in 2, which takes two input values and multiplies them to get an output value. If for all of the multiplication gates in a circuit, \\(\textbf{a}\_L\\) is the vector of left inputs, \\(\textbf{a}\_R\\) is the vector of right inputs, and \\(\textbf{a}\_O\\) is the vector of outputs, then the following equation will represent the relationship between the inputs and outputs of all the multiplication gates in the circuit:
+
+\\[
+\textbf{a}\_L \circ \textbf{a}\_R = \textbf{a}\_O
+\\]
+
+Other types of gates in arithmetic circuits are constant multiplication gates and addition gates. A constant multiplication gate can be represented by a constant multiplied by an input, and an addition gate can be represented by the addition of two or more inputs. All of the constant multiplication gates and addition gates in a circuit can be represented as a collection of linear constraints, as expressed in the following equation:
+
+\\[
+\textbf{W}\_L \cdot \textbf{a}\_L +
+\textbf{W}\_R \cdot \textbf{a}\_R +
+\textbf{W}\_O \cdot \textbf{a}\_O =
+\textbf{W}\_V \cdot \textbf{v} +
+\textbf{c}
+\\]
+
+Combining statements using challenge variables
+----------------------------------------------
+
+We can rewrite the statement about multiplication gates into an inner product equation, using the challenge variable \\(y\\). We can do this for a random challenge \\(y\\) because \\({\mathbf{b}} = {\mathbf{0}}\\) if and only
+if[^2] \\({\langle {\mathbf{b}}, {\mathbf{y}}^{n} \rangle} = 0\\). 
```

Comment:
Hmm, it looks like rustdoc doesn't do well with non-unique footnote numbers: clicking this takes you down to the bottom, but hitting the return link sends you back to the beginning of the RP notes.

Reply:
I fixed it up by having the footnotes be unique. It's currently awkward because there are two sets of footnotes that are the exact same, but when we split the notes into separate files (one for range proof, one for circuit proof, one for inner product proof) it make more sense.